### PR TITLE
Reporting documentation

### PR DESCRIPTION
**Pull request by edmundoa**
This PR contains all basics for the Reporting documentation. I still need to add documentation for plugin authors, but I guess that can be done separately. If you notice anything missing, please let me know and I can add it (or feel free of add it yourself).

I compressed the images as well, so they take a bit less space.

### CODE REVIEW

File: pages/faq.rst [hunk 1]
```diff
@@ -118,12 +118,12 @@ Yup, we’re all over this too with read/write roles and group permissions.  To
 Do we have a user audit log for compliance?
 -------------------------------------------
 
-`Graylog Enterprise <https://www.graylog.org/enterprise>`_ includes an audit log plugin. You can explore the :ref:`documentation <auditlog_toc>` for more details.
+`Graylog Enterprise <https://www.graylog.org/enterprise>`_ includes a audit log functionality. You can explore the :ref:`documentation <auditlog_toc>` for more details.
```
Comment (ChristopherKB) [inline pages/faq.rst]: remove the a between includes and audit. Should read Graylog Enterprise includes audit log functionality.

Comment (edmundoa) [reply pages/faq.rst]: Will do, thank you! 👍

**Review by jalogisch — CHANGES_REQUESTED** [review]
LGTM

Could you please add - possible at the schedule section that you need to configure the e-mail server in the `server.conf´ to have that working. 

I doubt that all people will have that.

**Comment by edmundoa** [comment]
@jalogisch that is already included in https://github.com/Graylog2/documentation/pull/554/files#diff-67751f6d416340c8a65ff3c884b90b52R67. I could make it a note to be more visible, but I guess there will be more notes than text on that section.

**Comment by jalogisch** [comment]
@edmundoa my bad - I overread that. Maybe that should be a `.. warning::`

**Comment by edmundoa** [comment]
Done!

File: pages/reporting/setup.rst [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+Installation
+============
+
+Reporting is part of the Graylog Enterprise plugin, please check the :doc:`Graylog Enterprise setup page </pages/enterprise/setup>`
+for details on how to install it.
+
+Configuration
+=============
+
+In most cases you can use the Reporting functionality without doing any changes in
```
Comment (ChristopherKB) [inline pages/reporting/setup.rst:17]: Should be ...without making any changes to...

File: pages/reporting/setup.rst [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+your Graylog configuration file (check the :ref:`default file locations page <default_file_location>`
+to see where you can find it). Next you can find all available configuration
```
Comment (ChristopherKB) [inline pages/reporting/setup.rst:19]: Change "Next you can find..." to "Below, you will find...

File: pages/reporting/setup.rst [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+----------------------
+Default value: ``false``.
+
+For security reasons, report generation will run in a sandbox and not as root
+user. There are two cases where you may consider enabling this option:
+
+- Environments where you want or must use the root user to run reporting
+  generation.
+- Environments that provide limited kernel capabilities, like a docker container.
+
```
Comment (ChristopherKB) [inline pages/reporting/setup.rst:49]: What would the impact be of disabling it? Is there performance gains or other reasons you might benefit or suffer from changing the default value?

File: pages/reporting/setup.rst [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+report_generation_timeout_seconds
+---------------------------------
+Default value: ``180``.
+
+Time in seconds to wait for a report to load in the background. You may
+need to increase this value if your reports are doing expensive queries
+or your ES cluster is a bit slow.
```
Comment (ChristopherKB) [inline pages/reporting/setup.rst:56]: How will they know when they need to increase it? Will reports time out or be otherwise affected if they are doing expensive queries?

File: pages/reporting/usage.rst [hunk 6]
```diff
@@ -0,0 +1,114 @@
+*****
+Usage
+*****
+
+Reports let you combine multiple Dashboard widgets to create a document that
+can display certain information in your Graylog system in an organized format
```
Comment (ChristopherKB) [inline pages/reporting/usage.rst:6]: change "...in your..." to "...contained within..."

File: pages/reporting/usage.rst [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+----------
+
+.. warning:: Please ensure the :ref:`email configuration <email_config>` in
+   your Graylog configuration file is working before you enable report
+   scheduling.
+
+.. note:: Scheduling Reports will use resources in the background on both your
+   Graylog and ES cluster. To avoid performance issues, make sure to allocate
+   enough resources for your setup and also disable scheduling of Reports you
+   don't need to be send automatically.
```
Comment (ChristopherKB) [inline pages/reporting/usage.rst:59]: change "send" to "sent".

File: pages/reporting/usage.rst [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+
+On the Scheduling section you can configure how often the report will be sent. It
```
Comment (ChristopherKB) [inline pages/reporting/usage.rst:61]: Change "On" to "In"

File: pages/reporting/usage.rst [hunk 9]
```diff
@@ -0,0 +1,114 @@
...
+is possible to send reports daily, weekly or on a monthly basis.
```
Comment (ChristopherKB) [inline pages/reporting/usage.rst:62]: Change "...reports daily, weekly or on a monthly basis." to "...reports on a daily, weekly or monthly basis."

File: pages/reporting/usage.rst [hunk 10]
```diff
@@ -0,0 +1,114 @@
...
+
+
+Layout
+------
+
+Much like in a Dashboard, you can drag and drop widgets on the virtual sheet of
+paper to select the orders the widgets should go in the report. Rearranging
+widgets will save the change in the layout automatically.
+
+Please note that the cover page will always go in the first page of the report, and
```
Comment (ChristopherKB) [inline pages/reporting/usage.rst:79]: Change "...will always go in the first page of the report" to "...will always be the first page of the report..."

File: pages/reporting/usage.rst [hunk 11]
```diff
@@ -0,0 +1,114 @@
...
+the next page will start with the report description followed by all widgets in
+the configured order.
+
+History
+=======
+
+As the background generation of reports may fail, the Report History page can help
+you finding out if there was some error while generating and sending a report in
```
Comment (ChristopherKB) [inline pages/reporting/usage.rst:87]: Change "you finding out if there was some error while generating..." to "you discover if there were any errors while generating..."

File: pages/reporting/usage.rst [hunk 12]
```diff
@@ -0,0 +1,114 @@
...
+the background.
+
+To open the Report history page for a report, click on the "More actions" button
+for that report, and select "Report history".
+
+.. image:: /images/reporting-history-page.png
+
+Generating report on demand
```
Comment (ChristopherKB) [inline pages/reporting/usage.rst:95]: Should that be "Generating Reports On Demand"?

It's a subsection heading, right?

File: pages/reporting/usage.rst [hunk 13]
```diff
@@ -0,0 +1,114 @@
...
+go to the "Enterprise/Reports" page, click on the "More actions" button for the
+report you want to download, and select "Download report now".
+
+Please take into account that the report generation may take a while.
+
+
+Send report as email manually
+-----------------------------
+
+Additionally to downloading a report on demand, you can also generate and send
```
Comment (ChristopherKB) [inline pages/reporting/usage.rst:111]: Change "Additionally to downloading...you can also..." to "In addition to downloading...you may also..."

**Review by ChristopherKB — COMMENTED** [review]
Looks great! Very few changes, most are colloquial or minor grammar changes.

**Review by ChristopherKB — COMMENTED** [review]
Looks good. Only minor changes suggested.

**Comment by edmundoa** [comment]
Thank you Chris! I will do another PR fixing those 👍

File: pages/reporting/usage.rst (hunk 12 above)
Comment (edmundoa) [reply pages/reporting/usage.rst:95]: That's right, changed it! 👍

File: pages/reporting/setup.rst (hunk 5 above)
Comment (edmundoa) [reply pages/reporting/setup.rst:56]: I'll rephrase it to make it clearer and more helpful, but due to the architecture of reporting, it's a difficult question to answer :)

File: pages/reporting/setup.rst (hunk 4 above)
Comment (edmundoa) [reply pages/reporting/setup.rst:49]: The impact is a somewhat lower level of security. The report generation is done by a headless Chrome browser, which by default runs in a sandbox that offers some extra security features. Disabling the sandbox is considered less safe, so we want to make people keep it on if possible.

I'll try to clarify that without adding too many gory details, which I also don't know that much.
